pkgs/development/libraries/libjson-rpc-cpp/default.nix

```diff
@@ -1,30 +1,69 @@
-{ stdenv, fetchurl, cmake, curl }:
+{ stdenv
+, fetchgit
+, cmake
+, jsoncpp
+, argtable
+, curl
+, libmicrohttpd
+, doxygen
+, catch
+}:
+stdenv.mkDerivation rec {
+  name = "libjson-rpc-cpp-${version}";
+  version = "0.6.0";
 
-let
-  basename = "libjson-rpc-cpp";
-  version = "0.2.1";
-in
+  src = fetchgit {
+    url = https://github.com/cinemast/libjson-rpc-cpp.git;
+    sha256 = "00fxxisg89zgg1wq047n8r8ws48jx35x3s6bbym4kg7dkxv9vv9f";
+    rev = "c6e3d7195060774bf95afc6df9c9588922076d3e";
+  };
 
-stdenv.mkDerivation {
-  name = "${basename}-${version}";
+  patchPhase = ''
+    for f in cmake/FindArgtable.cmake \
+             src/stubgenerator/stubgenerator.cpp \
+             src/stubgenerator/stubgeneratorfactory.cpp
+    do
+      sed -i -re 's/argtable2/argtable3/g' $f
+    done
 
-  src = fetchurl {
-    url = "https://github.com/cinemast/${basename}/archive/${version}.tar.gz";
-    sha256 = "1pc9nn4968qkda8vr4f9dijn2fcldm8i0ymwmql29h4cl5ghdnpw";
-  };
+    sed -i -re 's#MATCHES "jsoncpp"#MATCHES ".*/jsoncpp/json$"#g' cmake/FindJsoncpp.cmake
+  '';
+
+  configurePhase = ''
+    mkdir -p Build/Install
+    pushd Build
+
+    cmake .. -DCMAKE_INSTALL_PREFIX=$(pwd)/Install \
+             -DCMAKE_BUILD_TYPE=Release
+  '';
+ 
+  installPhase = '' 
+    mkdir -p $out
+
+    function fixRunPath {
+      p=$(patchelf --print-rpath $1)
+      q="$p:${stdenv.lib.makeLibraryPath [ stdenv.cc.cc jsoncpp argtable libmicrohttpd curl ]}:$out/lib"
+      patchelf --set-rpath $q $1
+    }
+
+    make install
 
-  buildInputs = [ cmake curl ];
+    sed -i -re "s#-([LI]).*/Build/Install(.*)#-\1$out\2#g" Install/lib/pkgconfig/*.pc
```

Comment:
Hey @artuuge! I want to update the `libjson-rpc-cpp` to `1.2.0`, I've started with this http://ix.io/1X0R (added `redis` and `hiredis` deps), but I'm not sure what else needs to be changed...

Reply:
I've removed that line and added 
```
-DUNIX_DOMAIN_SOCKET_SERVER=YES \
-DUNIX_DOMAIN_SOCKET_CLIENT=YES
```

Reply:
ok, got it working, I'll make a PR soon ;)